src/analysis/calculus/local_extr.lean

```diff
@@ -350,3 +351,28 @@ classical.by_cases
       let ⟨c, hc, hcdiff⟩ := h in ⟨c, hc, deriv_zero_of_not_differentiable_at hcdiff⟩)
 
 end Rolle
+
+namespace polynomial
+
+lemma card_root_set_le_derivative {F : Type*} [field F] [algebra F ℝ] (p : polynomial F) :
+  fintype.card (p.root_set ℝ) ≤ fintype.card (p.derivative.root_set ℝ) + 1 :=
+begin
+  haveI : char_zero F := char_zero_of_inj_zero
+    (λ n hn, by rwa [←(algebra_map F ℝ).injective.eq_iff, ring_hom.map_nat_cast,
+      ring_hom.map_zero, nat.cast_eq_zero] at hn),
```

Comment:
If we have `[field F] [field K] [algebra F K]` then `char_zero F` iff `char_zero K`, right? If this lemma doesn't exist yet, that would be nice to separate out and use here.